specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-07-01/managedClusters.json

```diff
@@ -1691,6 +1745,7 @@
           "description": "Version of orchestrator specified when creating the managed cluster."
         },
         "nodeImageVersion": {
+          "readOnly": true,
```

Comment:
why is this marked as read only? the POST call is going to change the value, correct?

Reply:
It will be a readonly value on 07-01 API version.
The POST call has no payload.

Reply:
some history context: initially we decide to use PUT to trigger the operation by input "nodeImageVersion: latest" and the returned value is something like "nodeImageVersion: AKSUbuntu-1804-2020.06.24". The main reason is to cover ARM template and terryform user. But that design make the API not idempotent and the internal implementation complicated. Thus we decide to change to use POST operation to trigger the node image version upgrade start by 2020-07-01. The POST operation will trigger the node image upgrade to the latest one. The latest node image version will be updated frequently (weekly). Customer cannot specify a specific version as we only support the latest version at the time of upgrading.

Reply:
but you are not solving the problem of being able to invoke this from a template. Has this change been made to the service side already? what will happen if the user rerun the PUT after calling the POST to upgrade? Is there a way to set this via the PUT/pATCH?

Reply:
we have discussed the template scenario, it will follow the existing solution we have for reset service principal, update add profile which are all using POST to trigger the operation.

Reply:
Customer cannot specify a specific version, that means they do not know what value to PATCH and PUT